amf-aml/shared/src/main/scala/amf/aml/internal/transform/steps/DialectReferencesResolutionStage.scala

```diff
@@ -162,14 +162,12 @@ class DialectReferencesResolutionStage() extends TransformationStep() {
 
   private def linkExtendedNodes(alreadyResolved: mutable.Map[String, NodeMappable]): Unit = {
     alreadyResolved.values.foreach { nodeMappable =>
-      nodeMappable.extend.headOption match {
-        case Some(extended: NodeMappable)
+      val extensions = nodeMappable.extend.map {
+        case extended: NodeMappable
             if extended.linkTarget.isDefined && alreadyResolved.contains(extended.linkTarget.get.id) =>
```

Comment:
Replace `extended.linkTarget.isDefined` for `extended.isLink`

Reply:
Also could extract `alreadyResolved.contains(extended.linkTarget.get.id)` into a variable

Reply:
Replaced the link function. The other cannot be extracted because is an inline if case condition